code/gateway/dataaccess.q

```diff
@@ -173,29 +209,73 @@ partdict:{[input]
     et:$[a;options`endtime;`date$options`endtime];
+
     // get the dates that are required by each process
-    dates:group key[part]where each{within[y;]each value x}[part]'[l:st+til 1+et-st];
+    dates:$[b;
+        {key[y]!where each flip x};
+        {group key[y]where each x}][;part]{within[y;]each value x}[part]'[l:st+til 1+et-st];
     dates:l{(min x;max x)}'[dates];
+
     // if start/end time not a date, then adjust dates parameter for the
     // correct types
     if[not a;
         // converts dates dictionary to timestamps/datetimes
         dates:$[-15h~tp;{"z"$x};::]{(0D+x 0;x[1]+1D-1)}'[dates];
         // convert first and last timestamp to start and end time
-        dates:@[dates;f;:;(start;dates[f:first key dates;1])];
-        dates:@[dates;l;:;(dates[l:last key dates;0];options`endtime)]];
+        $[b;
+            dates:key[dates]!?[value[dates][;0]<start;start;value[dates][;0]],'?[value[dates][;1]>end;end:options`endtime;value[dates][;1]];
+            [dates:@[dates;f;:;(start;dates[f:first key dates;1])];
+            dates:@[dates;l;:;(dates[l:last key dates;0];options`endtime)]]
+            ];
+        ];
+
     // adjust map reducable aggregations to get correct components
     if[(1<count dates)&`aggregations in key options;
         if[all key[o:options`aggregations]in key aggadjust;
             aggs:mapreduce[o;$[`grouping in key options;options`grouping;`]];
             options:@[options;`aggregations;:;aggs]]];
+
     // create a dictionary of procs and different queries
-    :{@[@[x;`starttime;:;y 0];`endtime;:;y 1]}[options]'[dates];
+    query:{@[@[x;`starttime;:;y 0];`endtime;:;y 1]}[options]'[dates];
+    // adjust query if striped
+    if[b&`instruments in key options;
+        modquery:select serverid,{x`skeysym`skeytime}each attributes from .gw.servers where({all`skeysym`skeytime in key x}each attributes)&serverid in raze key part;
+        
+        querytable:0!(`serverid xkey update serverid:(first each key query)from value query)uj`serverid xkey modquery;
+        // modify starttime, endtime and instruments based on stripe
+        querytable:update
+            {$[z;y;$[(stripest:x[1]0)<`time$y;y;stripest+`date$y]]}[;;a]'[attributes;starttime],
+            {$[z;y;$[(stripeet:x[1]1)<`time$y;stripeet+`date$y;y]]}[;;a]'[attributes;endtime],
+            // query instruments needs to be an atom if only 1sym is queried, if not it will throw a type error
+            adjinstruments:{$[1=count s:skeysym where(skeysym:.ds.stripe[(),y;x 0])in y;s 0;s]}'[attributes;instruments]
+                from querytable where serverid in modquery`serverid;
+
+        querytable:update adjinstruments:instruments from querytable where not serverid in modquery`serverid;
+        querytable:(enlist[`adjinstruments]!enlist `instruments)xcol enlist[`instruments]_querytable;
+        // filter queries not required
+        drops:exec serverid from querytable where 0=count each instruments;
```

Comment:
The reason for doing this is because I want to get the serverids to drop from the `key query` instead of `querytable` which is why I assigned it to the var `drops`. Here `key query` could be a nested list of `(1i;2i;3i;4 5i)` as well.

Reply:
Ah, OK in line 266 that is?

If `key query` can be a nested list, doesn't that `first each` mean that it'll only drop them if the first element from each nested list is in `drops`? e.g. with your example of `(1i;2i;3i;4 5i)`, wouldn't the last item be dropped if `drops` contains `4i`, but not if it only contains `5i`? Is that the expected behaviour?

Reply:
Yes that logic would be followed after line 250:

        querytable:0!(`serverid xkey update serverid:(first each key query)from value query)uj`serverid xkey modquery;

`querytable` was initialized using `first each key query`. So if the example `key query` was `(1i;2i;3i;4 5i)`, `5i` should not be present in `drops`.